### ISSUE TITLE
Unused warning in OSX debug compile for scrc.f90

### ISSUE THREAD

**Issue by rmcdermo** [firemodels/fds#6202]
@SusanKilian I keep getting this unused warning when compile OSX debug.  The AS variable is needed to compile, but maybe it is never used in an assignment?  Please check.
```
../../Source/scrc.f90(9202): remark #7712: This variable has not been used.   [AS]
TYPE (SCARC_MATRIX_TYPE), POINTER :: A, AS
----------------------------------------^
```

**Comment by SusanKilian** [comment]
Ah, I see, it's only used if WITH_MKL is defined (which I always use). Thanks for finding this. I suppose you compiled without MKL? In the future I'll test both. I'll make a corresponding PR ...

**Comment by rmcdermo** [comment]
I didn't intend to.  But I'm having compile issues at the moment.  So that's something I need to figure out.

**Comment by rmcdermo** [comment]
Fixed in PR #6203

**Comment by rmcdermo** [comment]
@SusanKilian I'm just getting my gfortran compilation to work again and if WITH_MKL is not defined, we get several more unused warnings.  Also, please make sure that you make a graceful exit if a user asks to use PARDISO preconditioner but has not compiled with MKL.  GLMAT has an example to follow.  Thanks.  Below are the errors we see:

```
../../Source/scrc.f90:821.72:

TYPE (SCARC_SOLVER_TYPE) , SAVE, TARGET :: COARSE_KRYLOV, COARSE_CLUSTER, COARSE_PARDISO
                                                                        1
Warning: Unused PRIVATE module variable 'coarse_cluster' declared at (1)
../../Source/scrc.f90:821.88:

TYPE (SCARC_SOLVER_TYPE) , SAVE, TARGET :: COARSE_KRYLOV, COARSE_CLUSTER, COARSE_PARDISO
                                                                                        1
Warning: Unused PRIVATE module variable 'coarse_pardiso' declared at (1)
../../Source/scrc.f90:820.84:

TYPE (SCARC_SOLVER_TYPE) , SAVE, TARGET :: MAIN_KRYLOV, MAIN_MULTIGRID, MAIN_CLUSTER
                                                                                    1
Warning: Unused PRIVATE module variable 'main_cluster' declared at (1)
../../Source/scrc.f90:822.113:

CARC_SOLVER_TYPE) , SAVE, TARGET :: PRECON_JACOBI, PRECON_SSOR, PRECON_FFT, PRECON_PARDISO, PRECON_CLUSTER, PR
                                                                                                          1
Warning: Unused PRIVATE module variable 'precon_cluster' declared at (1)
../../Source/scrc.f90:822.97:

TYPE (SCARC_SOLVER_TYPE) , SAVE, TARGET :: PRECON_JACOBI, PRECON_SSOR, PRECON_FFT, PRECON_PARDISO, PRECON_CLUS
                                                                                                 1
Warning: Unused PRIVATE module variable 'precon_pardiso' declared at (1)
../../Source/scrc.f90:128.26:

CHARACTER(40) :: SCARC_MKL       = 'GLOBAL'                 !> Type of MKL solver (LOCAL->Pardiso/GLOBAL->Clus
                          1
Warning: Unused PRIVATE module variable 'scarc_mkl' declared at (1)
../../Source/scrc.f90:129.32:

CHARACTER(40) :: SCARC_MKL_MTYPE = 'SYMMETRIC'              !> Type of MKL matrix (SYMMETRIC/UNSYMMETRIC)
                                1
Warning: Unused PRIVATE module variable 'scarc_mkl_mtype' declared at (1)
../../Source/scrc.f90:823.113:

CARC_SOLVER_TYPE) , SAVE, TARGET :: SMOOTH_JACOBI, SMOOTH_SSOR, SMOOTH_FFT, SMOOTH_PARDISO, SMOOTH_CLUSTER
                                                                                                          1
Warning: Unused PRIVATE module variable 'smooth_cluster' declared at (1)
../../Source/scrc.f90:823.97:

TYPE (SCARC_SOLVER_TYPE) , SAVE, TARGET :: SMOOTH_JACOBI, SMOOTH_SSOR, SMOOTH_FFT, SMOOTH_PARDISO, SMOOTH_CLUS
                                                                                                 1
Warning: Unused PRIVATE module variable 'smooth_pardiso' declared at (1)
../../Source/scrc.f90:5306:0: warning: ‘scarc_setup_ludecomp’ defined but not used [-Wunused-function]
 SUBROUTINE SCARC_SETUP_LUDECOMP(NSOLVER, NSCOPE, NSTACK, NLMIN, NLMAX)
 ^
```

**Comment by SusanKilian** [comment]
Hopefully, this problem is solved now.  The code should now gracefully exit if Pardiso or Cluster_Sparse_Solver are specified in the input file and the MKL library is not compiled.

I will continue to improve my messaging mechanism in the medium term ...

**Comment by rmcdermo** [comment]
Solved in PR #6207
